### DISCUSSION TITLE
Logueo como Administrador o Usuario - Abogabot

### DISCUSSION THREAD

**Discussion by AlbertoSF99** [Launch-X-Latam/MisionFrontEnd#154] [Q&A]
Buenos días/tardes/noches compañeros. Me surgió una duda con respecto a la funcionalidad del Abogabot. Tengo entendido que el sitio web/aplicación deberá tener un Dashboard diferente dependiendo si es el abogado que está haciendo uso de este (para visualizar el pago, el progreso del caso, etc.) o en caso de que el cliente del abogado esté ocupando este servicio (para desplegar el formulario, opciones de pago, seguimiento, etc.). Ahora bien, mi duda es, ¿Cómo hago para desplegar "x" o "y" Dashboard dependiendo de si se trata de un administrador (en este caso, el abogado) o usuario (el potencial cliente del abogado)? Tenía la idea de hacer el típico sistema de logueo, pero uno para administrador y otro para usuario, así pues, se podría loguear o crear una cuenta aquel que cumpla los datos o requerimientos para poder usar el Dashboard de administrador (abogado, disculpen si "administrador" no es la palabra correspondiente, no sabía que término usar para él) pero esta opción se me hace un poco inseguro y no muy viable, puesto que debería verificarse que esta persona cumpla con el perfil de abogado que preste sus servicios. ¿Alguna idea de como podría hacer esta conceptualización del logueo dependiendo de si se trata del abogado o del cliente de este abogado? Y una disculpa de antemano si no se logra entender bien la pregunta, escribí esto mientras capturaba las ideas en mi cabeza. Saludos.

**Comment by DaveOval** [comment]
Yo creo que sería el mismo sistema de inicio solo que tal vez al momento de entrar a la plataforma debería de usar un correo de la empresa. 
Ejemplo un usuario normal entraría con un correo @gmail.com mientras qué un admin debería de entrar con un correo empresarial como @abogabot.net o algo así y en ese momento ya se despliega el menú correspondiente para cada usuario. Saludos y espero haberte ayudado.

**Reply by EduGarcia502** [reply]
exactamente concuerdo contigo, desde el cpanel puedes darle privilegios a ciertos usuarios, por ejemplo determinar quien es cliente y quien es administrador, pero eso ya es cosa aparte

**Reply by AlbertoSF99** [reply]
Ooh, tiene sentido lo que dices, así automáticamente se hace distinción de si el usuario es el abogado o el cliente con las credenciales de usuario del mismo, y con esto, se puede desplegar la información requerida más fácilmente. Muchas gracias.

**Comment by iam-leo** [comment]
Creo que esto te puede ayudar

![maxresdefault](https://user-images.githubusercontent.com/79548542/194431230-4d441f7d-0e70-4192-9749-6886e2f767f2.jpg)

Yo lo veo como el caso B que se muestra en la imagen, puedes hacer solo un login, como menciona @DaveOval, para ambos tipos de usuarios y luego con un condicional verificar por ej si es administrador muestras un dashboard y si no lo es muestras otro.

En cuanto a la cuenta del administrador ya le puedes entregar directamente sus credenciales y cuando entre la primera vez obligarle a cambiar la contraseña, pero siempre utilizando el mismo login, es un ejemplo, tu lo puedes diagramar como te lo imagines.

**Reply by AlbertoSF99** [reply]
Entiendo, por poner una analogía, es más o menos con lo que sucede con nuestros correos institucionales de Microsoft, ya que al loguearnos con estas credenciales del programa Launch X redirecciona a una página con un Dashboard diferente al que un usuario común tendría con su correo particular. Gracias por la aclaración amigo.

**Reply by iam-leo** [reply]
Exacto, ese es un modelo que puedes utilizar o puedes pensar otra lógica, lo bueno de todo esto es que hay muchas formas de realizarlo y cada uno se ajustará a lo que crea conveniente. 
Espero que te haya servido la ayuda compa!

**Comment by VictorFalconi** [comment]
Podrias implementar un registro para abogados y otro para clientes. En muchas paginas es posible registrarse segun el rol que cumples en la institución. Ya si es que quieres verificar que efectivamente es un abogado para algo mas serio, tendrias que acceder a alguna base de datos de abogados de tu pais, alguna institucion que reuna a todos los abogados. Aqui en Peru tenemos el colegio de abogados. Tendrian que pasar la validacion antes de tener cuenta activa

**Reply by Ayuya2001** [reply]
yo tome el ejemplo como que abogabot es una pagina que ayuda a un grupo de abogados, no a todos los abogados del pais en cuestión, a un buffet de abogados por ej

**Comment by kerizrh** [comment]
Yo lo había de entendido de otra manera. Creí que los administradores de la app simplemente tendrían una opción (ventana o escritorio) diferente dónde tendrían los permisos de administrador (valga la redundancia) para entrar en la app dónde podrían ver lo que es la notificación de pago sin tener que diseñar un UI o UX completamente diferente para ellos.  Y que estos ya deberían de estar controlados desde los dispositivos propios de una empresa en específico.  No sé qué tan perdido estoy o si se entiende mi idea.

**Reply by AlbertoSF99** [reply]
No estoy seguro si se deban hacer dos UI/UX diferentes e independientes entre si, pero por lo que entendí, dependiendo de quien se loguee se tendrán ventanas y Dashboards diferentes (para el caso del cliente, el formulario y el apartado de seguimiento de su caso, y para el administrador/abogado el Dashboard de los pagos y las alertas/notificaciones de los casos nuevos).

**Reply by maxkario** [reply]
Si es posible un UX y UI. Allí deberíamos, como desarrolladores, crear una página para el perfil de usuario y otra para el administrador o encargado de revisar los pagos y demás ámbitos administrativos de la empresa. El diseño aplica para ambos casos.

Por otra parte, tenía la misma duda que planteas en el post. Saludos.

**Comment by eldhaim** [comment]
Ese tema es de backend compañero, debe hacer las 2 vistas y en la logica interna que se aplique más adelante puede incluir 2 estructuras (tablas), una para los rollers (ejemplo: cliente, admin, operador,etc...) Que contendrá una columna para relacionar en la otra tabla que sería la de roles. Entonces dependiendo del roll que cumpla el usuario que ingresa, es la vista que se despliega, en pocas palabras desde el mismo login habrá despliegue de las necesidades del usuario
